300-CNAB-security.md

```diff
@@ -0,0 +1,163 @@
+- *Layout:* A declaration of a series of attestations that must be made in order to consider a particular bundle to have satisfied the acceptance criteria.
+- *Link:* An envelope that binds a particular attestation to a particular step in a layout.
+- *Signature:* A cryptographic verification that the possessor of a particular private key has approved a particular body of content. For example, a signature may capture the approval: "Developer Padma has approved this archive file". Note that an attestation always contains a signature, but a signature may be used for things other than attestation.
+- *Verification:* The process of applying a public key to a signature and associated body of content, with the purpose of verifying that the body of content is identical to the body that was signed by the private key.
+
+## Securely Moving Bundles To And From Registries
+
+Per the CNAB Core 1.0 specification, bundles may be stored and transmitted in two forms: thin bundles and thick bundles. Both thin and thick bundles can be transmitted over the network. This section describes how CNAB clients (including CNAB runtimes) may safely establish the identity and integrity of a CNAB bundle being fetched from a CNAB registry.
+
+(Per the [CNAB Registry specification](200-CNAB-registries.md), CNABs may be stored in any OCI Registry 1.0-compliant implementation)
+
+Securing a CNAB registry is done by implementing the TUF (The Update Framework) specification on both the client and registry side, and then adhering to the protocol. A registry MUST comply with the [TUF 1.0 specification](https://github.com/theupdateframework/specification/blob/master/tuf-spec.md) to meet this specification. A client also MUST comply with the TUF specification to meet this specification. Furthermore, the client MUST perform additional verification on the content of a bundle.
+
+This specification does not require that a bundle repository support both thick and thin bundles. It prescribes processes for storage of each. Unless otherwise noted, a registry is considered _compliant_ if it implements storage for at least one of the two.
+
+This specification does not specifically deal with network-layer (Layer 7) security, but assumes that implementations will apply correct security constraints for all network traffic. TLS/SSL is likely the method by which such traffic should be secured.
+
+### Overview
+
+There are three major aspects of securely working with registries:
+
+1. Pulling bundles _from_ a registry _to_ a client
+2. Pushing bundles _from_ a client _to_ a registry
+3. Managing keys between a client and a registry
+
+### Pulling and Verifying Bundles
+
+When a client pulls a bundle from a repository, the client MUST verify the package, where verification consists of the following:
```

Comment:
do you mean to use the term repository and registry interchangeably?

Reply:
🤦‍♂️ No. I'll fix that. Old habits die hard.